common/src/main/java/io/crate/types/TimestampType.java

```diff
@@ -29,44 +35,115 @@
 import java.time.format.ResolverStyle;
 import java.time.temporal.TemporalAccessor;
 import java.util.Locale;
+import java.util.function.Function;
 
 import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
 import static java.time.format.DateTimeFormatter.ISO_LOCAL_TIME;
 
-public final class TimestampType extends BaseTimestampType {
+public final class TimestampType extends DataType<Long>
+    implements FixedWidthType, Streamer<Long> {
 
-    public static final TimestampType INSTANCE = new TimestampType();
-    public static final int ID = 15;
+    public static final int ID_WITH_TZ = 15;
```

Comment:
looks a bit weird with these constants, but we are forced to have them when we decide on data type based on its in in switch statements

Reply:
Would it be better if we keep them private (hide them) and use instance.id() instead?

Reply:
`switch` requires constants and doesn't work on `id()`. 

I don't see a problem exposing them like this.